bin/add-deadlines-to-legacy-tasks

```diff
@@ -36,20 +38,27 @@ const patchTaskDeadline = async (taskId, transaction) => {
         return {}; // submission was not complete and correct, reset deadline
       }
     }
-    if (activity.eventName === 'update') { // possible deadline extend event
+    if (activity.eventName === 'update' && !isEmpty(deadline)) { // possible deadline extend event
       deadline.isExtended = get(activity, 'event.data.extended', false);
       deadline.isExtendable = !deadline.isExtended;
     }
     return deadline;
   }, {});
 
   if (isEmpty(deadline)) {
-    console.log(`${task.id}: no deadline required`);
-    return Promise.resolve();
+    if (task.data.deadline) {
+      stats.removed++;
+      return task.$query(transaction).patchAndFetch({ data: { ...task.data, deadline: null } }).context({ preserveUpdatedAt: true });
```

Comment:
This bit is here to remove the partial deadlines (e.g. `{"isExtended":false,"isExtendable":true}`) that got added to the task records on preprod the first time we ran the script.

Reply:
Does this script backfill all tasks that need a deadline?

If so then I'd probably get rid of any code that handles preprod being wonky and just run a manual query to remove all deadlines from preprod.

Reply:
yeah it does, ok let's do that.

Reply:
done.